server/launcher/server.go

```diff
@@ -46,7 +47,9 @@ func New(
 func (hgprc *Handler) Handler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if r.ProtoMajor == 2 && strings.Contains(r.Header.Get("Content-Type"), "application/grpc") {
-			hgprc.ServeHTTP(w, r)
+			ctx := r.Context()
+			ctx = kithttp.PopulateRequestContext(ctx, r)
```

Comment:
why not

```
ctx := kithttp.PopulateRequestContext(r.Context(), r)
```

Reply:
looked cleaner to me to separate it on two lines, but I don't have a strong preference here.